src/graph/unit_graph.cc

```diff
@@ -1259,6 +1277,37 @@ HeteroGraphPtr UnitGraph::CopyTo(HeteroGraphPtr g, const DGLContext& ctx) {
   }
 }
 
+HeteroGraphPtr UnitGraph::PinMemory() {
+  CSRPtr pinned_in_csr, pinned_out_csr;
+  COOPtr pinned_coo;
+  if (this->in_csr_->defined() && !this->in_csr_->IsPinned()) {
```

Comment:
nit, let's avoid **if (not ...) {...} else {...}**, flip it if possible.
which means: swap condition 1 and 2.

if (this->in_csr_->defined() && this->in_csr_->IsPinned()) {
} else if (this->in_csr_->defined()) {
} else {
}

same for the others.